### PR TITLE
feat: remove through2 usage in table.ts

### PR DESCRIPTION
**Pull request by crwilcox**
Towards #708

### CODE REVIEW

**Comment by crwilcox** [comment]
Unit and System tests are good. Though two acceptance tests are failing
```
1) Read Row Acceptance tests
       invalid - duplicate row key:

      Uncaught AssertionError [ERR_ASSERTION]: row count mismatch
      + expected - actual

      -0
      +1

      at verify (build/system-test/read-rows-acceptance-tests.js:110:24)
      at PassThrough.table.createReadStream.on.err (build/system-test/read-rows-acceptance-tests.js:100:17)
      at Pumpify.rowStream.on (build/src/table.js:420:32)
      at Pumpify.Duplexify._destroy (node_modules/duplexify/index.js:195:15)
      at /Users/crwilcox/workspace/googleapis/nodejs-bigtable/node_modules/duplexify/index.js:185:10
      at process._tickCallback (internal/process/next_tick.js:61:11)

  2) Read Row Acceptance tests
       invalid - new row missing row key:

      Uncaught AssertionError [ERR_ASSERTION]: row count mismatch
      + expected - actual

      -0
      +1

      at verify (build/system-test/read-rows-acceptance-tests.js:110:24)
      at PassThrough.table.createReadStream.on.err (build/system-test/read-rows-acceptance-tests.js:100:17)
      at Pumpify.rowStream.on (build/src/table.js:420:32)
      at Pumpify.Duplexify._destroy (node_modules/duplexify/index.js:195:15)
      at /Users/crwilcox/workspace/googleapis/nodejs-bigtable/node_modules/duplexify/index.js:185:10
      at process._tickCallback (internal/process/next_tick.js:61:11)


```

**Comment by crwilcox** [comment]
cc: @bcoe

File: src/table.ts [hunk 1]
```diff
@@ -810,33 +810,34 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
...
-          next(null, row);
-        }),
-      ]);
+      // TODO: These lines are the changes that cause test failures.
+      // const transform = new Transform({objectMode: true});
+      // transform._transform = (rowData, _, next) => {
+      const transform = through.obj((rowData, _, next) => {
+        if (
+          chunkTransformer._destroyed ||
+          !userStream.writable
```
Comment (stephenplusplus) [inline src/table.ts:819]: Previously the line was "(userStream as any)._writableState.ended". Does this have any effect on the test failure?

Comment (crwilcox) [reply src/table.ts:819]: None :) I did this because on the internal streams writable seems to have the correct behavior.

**Comment by crwilcox** [comment]
After simplifying the mock for the acceptance tests one failure remains:
```

  1) Read Row Acceptance tests
       invalid - duplicate row key:

      Uncaught AssertionError [ERR_ASSERTION]: row count mismatch
      + expected - actual

      -0
      +1

      at verify (build/system-test/read-rows-acceptance-tests.js:110:24)
      at PassThrough.table.createReadStream.on.err (build/system-test/read-rows-acceptance-tests.js:100:17)
      at Pumpify.rowStream.on (build/src/table.js:428:32)
      at Pumpify.Duplexify._destroy (node_modules/duplexify/index.js:195:15)
      at /Users/crwilcox/workspace/googleapis/nodejs-bigtable/node_modules/duplexify/index.js:185:10
      at process._tickCallback (internal/process/next_tick.js:61:11)

```

**Comment by JustinBeckwith** [comment]
Trying to clean up cruft PRs, so letting this one go

**Pull request closed without merging**